runhouse/resources/hardware/launcher_utils.py

```diff
@@ -232,11 +260,12 @@ def up(self, verbose: bool = True):
                 )
             raise e
 
-    def teardown(self, verbose: bool = True):
+    @classmethod
+    def teardown(cls, cluster, verbose: bool = True):
         """Tearing down a cluster locally via Sky."""
         import sky
 
-        sky.down(self.cluster.name)
+        sky.down(cluster.name)
 
```

Comment:
maybe we keep a similar structure to `up`, where for the den launcher we handle the saving in Den, and for the local launcher we add this after calling `sky down`: 

```
        # Save to Den with updated null IPs
        if rns_client.autosave_resources():
            self.save()
```